Fix #11: Introduce Oppia splash screen.

veena14cs opened the pull request:
**Completed**
Created splash screen.
Added splash logo.

**Check list**
- [x]  Faded transition when loading home page

- [x] Handling Configuration changes

veena14cs committed:
created splash screen

veena14cs committed:
changed splash image

veena14cs committed:
added fade transition

veena14cs committed:
handle configuration changes

veena14cs committed:
created background for splashtheme

This is to avoid black or white screen when app is launched and resources are not yet initialized

veena14cs committed:
implemented Logging system

- Used hyperLog library to store app logs in local file.
- Logger class is used to log files in entire application
- OppiaApplication class is created to initialize hyperlog

BenHenning commented:
@veena14cs can you please extract the logging API introduction to its own pull request? We should keep these pull requests as focused as possible to make them easier to review.

BenHenning commented on app/src/main/AndroidManifest.xml:
```diff
@@ -7,18 +7,26 @@
     android:allowBackup="true"
     android:icon="@mipmap/ic_launcher"
     android:label="@string/app_name"
+    android:name="com.example.myapplication.OppiaApplication"
```
Why do we need this?

veena14cs replied:
Its used for logging system, to initialize hyperlog I created application class which is initialized only once throughout the app.

BenHenning commented on app/src/main/AndroidManifest.xml:
```diff
@@ -7,18 +7,26 @@
     android:supportsRtl="true"
     android:theme="@style/OppiaTheme">
-    <activity android:name="org.oppia.app.HomeActivity">
+    <activity android:name=".HomeActivity">
```
Isn't this already declared in this manifest?

veena14cs replied:
its been removed.

BenHenning commented on app/src/main/AndroidManifest.xml:
```diff
@@ -7,18 +7,26 @@
+    <activity android:name=".HomeActivity">
+    </activity>
+    <activity android:name=".splash.SplashActivity"
+              android:configChanges="orientation|screenSize|keyboardHidden"
```
Why do we want to override the default configuration handling for this activity?

veena14cs replied:
To avoid activity restart it was added. Now I have done changes to classes so removed this default configuration

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashActivity.kt:
```diff
@@ -0,0 +1,27 @@
+package org.oppia.app.splash
+
+
```
Nit: please remove extra space.

veena14cs replied:
removed extra spaces

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashActivity.kt:
```diff
@@ -0,0 +1,27 @@
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_splash)
+
```
Ditto here. For consistency, let's use one space between blocks of code when separating things out.

Ditto elsewhere in this PR.

veena14cs replied:
removed extra spaces all over

BenHenning commented on .idea/misc.xml:
```diff
@@ -6,4 +6,4 @@
     <option name="id" value="Android" />
   </component>
-</project>
+</project>
```
Can you revert this? Let's keep EOF line endings for consistency.

veena14cs replied:
its resolved

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashActivity.kt:
```diff
@@ -0,0 +1,27 @@
+        val fragment = SplashFragment()
+        val transaction = supportFragmentManager.beginTransaction()
+        transaction.replace(R.id.fragment_container, fragment)
+        transaction.commit()
```
Prefer ``commitNow()`` since ``commit()`` can cause issues when your activity undergoes configuration or lifecycle changes mid-commit (resulting in state loss).

veena14cs replied:
commitNow() is used instead of commit().

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashActivity.kt:
```diff
@@ -0,0 +1,27 @@
+import com.example.myapplication.SplashFragment
+import org.oppia.app.R
+
+class SplashActivity : AppCompatActivity() {
```
Please add a Javadoc comment explaining what this activity is. Ditto for the fragment below.

veena14cs replied:
Added JavaDoc comments.

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashFragment.kt:
```diff
@@ -0,0 +1,69 @@
+
+        // TODO(Veena): Replace below code to load profile data and then redirect to HomePage
+
+        Handler().postDelayed(// Using handler with postDelayed called runnable run method
```
I'm still concerned with this--we shouldn't be handling configuration changes for this activity, and removing that line may result in a crash if the user rotates during the transition.

Why do we need this wait? Can't we just immediately transition to the home activity?

veena14cs replied:
if we don't add wait time, transition is too quick. Will that be ok.? As it is setting background theme in the splashActivity. Splash screen will be shown until app resources are initialized which is hardly 1 to 2 seconds

veena14cs replied:
I have removed handler and now there is transition without delay

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashFragment.kt:
```diff
@@ -0,0 +1,69 @@
+        // close this activity
+
+        activity!!.finish()
+        activity!!.overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
```
Can this be declared at the manifest level?

veena14cs replied:
yes. I could define it in styles and apply that theme to activity.

BenHenning commented on app/src/main/res/anim/fade_in.xml:
```diff
@@ -0,0 +1,5 @@
+<alpha xmlns:android="http://schemas.android.com/apk/res/android"
+       android:interpolator="@android:anim/accelerate_interpolator"
+       android:fromAlpha="0.0" android:toAlpha="1.0"
+       android:duration="500" />
```
I recommend moving the durations to dimens.xml so that we can keep the fade-in/fade-out times consistent here & below.

veena14cs replied:
moved durations to dimens.xml

BenHenning commented on app/src/main/res/drawable/bg_splash.xml:
```diff
@@ -0,0 +1,10 @@
+
+  <item android:drawable="@color/colorPrimary" />
+
+  <!--<item android:drawable="@drawable/ic_oppia_logo_white"-->
```
Why is this commented out? Shouldn't we be using the logo?

veena14cs replied:
made changes to this file. I have added splash_page as background image

BenHenning commented on app/src/main/res/drawable/ic_oppia_icon_mint.xml:
```diff
@@ -0,0 +1,4 @@
+<vector android:height="24dp" android:viewportHeight="164.37"
```
Is 24dp the right size? Ditto below for other imported SVGs.

veena14cs replied:
I have removed unwanted icons.  Chantel has provided splash page with logo. That works fine.

BenHenning commented on app/src/main/res/layout/activity_splash.xml:
```diff
@@ -0,0 +1,19 @@
+    android:id="@+id/fragment_container"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    app:layout_constraintBottom_toBottomOf="parent"
```
Is constraint layout needed here? Maybe just use a FrameLayout parent for simplicity.

veena14cs replied:
removed constraint layout for simplicity

BenHenning commented on app/src/main/res/layout/fragment_splash.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+             android:background="@color/colorPrimary"
```
Nit: Please reformat this code for consistent spacing.

veena14cs replied:
Code reformatting done

BenHenning commented on app/src/main/res/layout/fragment_splash.xml:
```diff
@@ -0,0 +1,20 @@
+      android:layout_gravity="center"
+      android:layout_marginBottom="100dp"/>
+
+</FrameLayout>
```
Similar to above, please keep EOF line endings for consistency.

veena14cs replied:
done

BenHenning reviewed (changes requested):
Thanks @veena14cs! Took a first pass on it.

Besides moving the logging code to its own PR, I had one question about the imported SVGs: are we using all 4 of them here?

veena14cs committed:
fixed issues

veena14cs committed:
Update misc.xml

BenHenning commented:
One other suggestion: can you please add tests for this new splash activity/fragment? They should be AndroidJUnit tests such that they can run on both Robolectric & Espresso (similar to the HomeActivity's test), and should verify that the transition works as expected.

veena14cs commented:
> Thanks @veena14cs! Took a first pass on it.
> 
> Besides moving the logging code to its own PR, I had one question about the imported SVGs: are we using all 4 of them here?

- We are not using all of them. unwanted SVGs are removed.

veena14cs committed:
Fixed issues

- Removed unwanted spaces
- Removed wait time during transition
- Added Javadoc comment
- created background image for splashActivity and it is set to theme.
- Added animation to SplashTheme styles.

veena14cs commented:
@BenHenning PTAL

veena14cs committed:
working on test cases

droidizer commented:
@veena14cs There's a conflict in a file. :)

droidizer commented:
I happen to think it is better to introduce Dependency Injection already. @BenHenning  Thoughts? 
<strike>Also since the classes are quite simple how about using Kotlin?</strike>
Update: My bad!

BenHenning commented:
Agreed @droidizer. I'm working on introducing Dagger 2 as part of #4. There will be a PR for that soon. Regarding Kotlin--which files in this PR aren't using Kotlin? They all seem to be.

@veena14cs can you please resolve the conflict and update your branch?

BenHenning commented on .idea/misc.xml:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Please revert this change--I notice it keeps happening to me as well, but I've found one way to potentially fix it that will be checked into my next PR to resolve #4.

BenHenning commented on app/src/main/AndroidManifest.xml:
```diff
@@ -10,15 +10,19 @@
       </intent-filter>
     </activity>
+
+    <activity android:name=".HomeActivity">
```
I believe this can just be:

```xml
<activity android:name=".HomeActivity" />
```

To shorten things.

BenHenning commented on app/src/main/java/org/oppia/app/HomeActivity.kt:
```diff
@@ -33,7 +33,7 @@ class HomeActivity : AppCompatActivity() {
       override fun onFailure(t: Throwable) {
         // TODO(BenHenning): Replace this log statement with a clearer logging API.
-        Log.e("HomeActivity", "Failed to retrieve user app history", t)
+        Log.e("HomeActivity", "Failed to retrieve user app history = "+ t)
```
Why make this change? If it's related to the previous logging portion of this PR, please revert.

BenHenning commented on app/build.gradle:
```diff
@@ -60,6 +60,8 @@ dependencies {
     androidTestImplementation "com.google.truth:truth:0.43"
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
 
+    implementation 'com.hypertrack:hyperlog:0.0.10'
```
This is no longer necessary for the PR--right? If so, please revert.

veena14cs replied:
no its been resolved

veena14cs commented:
Sure, Will check and resolve conflict

On Sun, Aug 25, 2019 at 8:46 PM Ben Henning <notifications@github.com>
wrote:

> Agreed @droidizer <https://github.com/droidizer>. I'm working on
> introducing Dagger 2 as part of #4
> <https://github.com/oppia/oppia-android/issues/4>. There will be a PR for
> that soon. Regarding Kotlin--which files in this PR aren't using Kotlin?
> They all seem to be.
>
> @veena14cs <https://github.com/veena14cs> can you please resolve the
> conflict and update your branch?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia-android/pull/60?email_source=notifications&email_token=AD4LXZAYZ5JNMSBUZSJBMATQGNG2NA5CNFSM4IOCA73KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5DFWJY#issuecomment-524704551>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD4LXZE5BCPPGL7CR7B32MDQGNG2NANCNFSM4IOCA73A>
> .
>

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashActivity.kt:
```diff
@@ -0,0 +1,27 @@
+import android.content.res.Configuration
+
+/*
+* This activity acts as a container to SplashFragment
```
Nit: please keep sentences next to each other & align the ``*``s. Also, Javadocs begin with ``/**``. Finally, we should describe what the activity contains, rather than making it related to the fragment (since the fragment is really just an implementation detail). Suggested change:

```kotlin
/**
 * An activity that shows a temporary loading page until the app is fully opened.
 */
```

Though, if there are no additional details to include that that can be simplified to:

```kotlin
/** An activity that shows a temporary loading page until the app is fully opened. */
```

veena14cs replied:
Ok got it.:)

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashActivity.kt:
```diff
@@ -0,0 +1,27 @@
+class SplashActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
```
Although we don't yet have a style guide outlined in the overall architecture document, let's follow a hybrid of JetBrain's Kotlin style guide and Google's Java style guide. In this case, per https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation let's use 2 spaces for indentation and 4 spaces for continuation (which I think I configured at the project level for Kotlin--maybe make sure that you're using the project configuration for Kotlin in your IDE and try reformatting the code via the IDE).

We can evaluate later whether this is the spacing that we want, though I'm inclined to try and fit more code on a line rather than increasing whitespace since it makes it harder to read on smaller screens.

BenHenning replied:
This appears unresolved--can you please fix the spacing here & elsewhere in this PR?

veena14cs replied:
Fixed indentation

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashActivity.kt:
```diff
@@ -0,0 +1,27 @@
+
+        val fragment = SplashFragment()
+        val transaction = supportFragmentManager.beginTransaction()
+        transaction.replace(R.id.fragment_container, fragment)
```
Small nit: these four lines can be done in one statement. Let's do that instead of splitting it up.

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashFragment.kt:
```diff
@@ -0,0 +1,45 @@
+import org.oppia.app.R
+
+/*
+* The SplashFragments navigates to Home page once
```
This should be phrased in terms of its type, e.g.:

``A fragment that navigates to HomeActivity once...``

Please update, and format the Javadoc bits like mentioned above in HomeActivity.

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashFragment.kt:
```diff
@@ -0,0 +1,45 @@
+* The SplashFragments navigates to Home page once
+* the app is finished loading completely
+* */
+
```
Nit: Javadocs should not have a blank space between them and the class/method/field that they're documenting.

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashFragment.kt:
```diff
@@ -0,0 +1,45 @@
+        super.onCreate(savedInstanceState)
+
+    }
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
```
Nit: please add spacing between methods.

BenHenning replied:
Nit on continuation: prefer continuing at the higher syntactical element (e.g. '(' rather than ',') per https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break.

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashFragment.kt:
```diff
@@ -0,0 +1,45 @@
+    }
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        // Inflate the layout for this fragment
```
For this & the next comment (``close this activity``). The code seems pretty self-documenting in both places, so these comments seem unnecessary.

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashFragment.kt:
```diff
@@ -0,0 +1,45 @@
+    }
+
+}
+
```
Nit: let's have just one EOF line terminator. Ditto elsewhere.

BenHenning commented on app/src/main/res/layout/activity_splash.xml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
```
I think you mentioned earlier that this was changed away from a ConstraintLayout, but it seems like it's still one--can we make this a FrameLayout, instead?

veena14cs replied:
yes, its resolved

BenHenning commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <integer name="fade_in_duration">1000</integer>
```
Please include the time unit in the name, e.g. ``fade_in_duration_ms`` since this appears to be in milliseconds.

veena14cs replied:
Okay

veena14cs committed:
resoilved conflicts

BenHenning commented on app/src/main/res/values/styles.xml:
```diff
@@ -5,4 +5,25 @@
+  <style name="SplashScreenTheme" parent="OppiaThemeWithoutActionBar">
+    <item name="android:windowBackground">@drawable/bg_splash</item>
+    <item name="android:windowAnimationStyle">@style/Animation.Activity</item>
+    <item name="android:backgroundDimEnabled">true</item>
```
Out of curiosity, why do we want to enable this?

BenHenning reviewed (changes requested):
Thanks @veena14cs! Generally, the PR looks good to me. I had a few substance comments, but most were nits on style since we're in the process of establishing style guidelines.

veena14cs commented:
I forgot to remove it. I was trying some other transition effects.

On Sun, Aug 25, 2019 at 9:22 PM Ben Henning <notifications@github.com>
wrote:

> *@BenHenning* requested changes on this pull request.
>
> Thanks @veena14cs <https://github.com/veena14cs>! Generally, the PR looks
> good to me. I had a few substance comments, but most were nits on style
> since we're in the process of establishing style guidelines.
> ------------------------------
>
> In .idea/misc.xml
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317442318>:
>
> > @@ -1,6 +1,6 @@
>  <?xml version="1.0" encoding="UTF-8"?>
>  <project version="4">
> -  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
>
> Please revert this change--I notice it keeps happening to me as well, but
> I've found one way to potentially fix it that will be checked into my next
> PR to resolve #4 <https://github.com/oppia/oppia-android/issues/4>.
> ------------------------------
>
> In app/src/main/AndroidManifest.xml
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317442968>:
>
> >        </intent-filter>
>      </activity>
> +
> +    <activity android:name=".HomeActivity">
>
> I believe this can just be:
>
> <activity android:name=".HomeActivity" />
>
> To shorten things.
> ------------------------------
>
> In app/src/main/java/org/oppia/app/HomeActivity.kt
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317443074>:
>
> > @@ -33,7 +33,7 @@ class HomeActivity : AppCompatActivity() {
>
>        override fun onFailure(t: Throwable) {
>          // TODO(BenHenning): Replace this log statement with a clearer logging API.
> -        Log.e("HomeActivity", "Failed to retrieve user app history", t)
> +        Log.e("HomeActivity", "Failed to retrieve user app history = "+ t)
>
> Why make this change? If it's related to the previous logging portion of
> this PR, please revert.
> ------------------------------
>
> In app/build.gradle
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317443107>:
>
> > @@ -60,6 +60,8 @@ dependencies {
>      androidTestImplementation "com.google.truth:truth:0.43"
>      androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
>
> +    implementation 'com.hypertrack:hyperlog:0.0.10'
>
> This is no longer necessary for the PR--right? If so, please revert.
> ------------------------------
>
> In app/src/main/java/org/oppia/app/splash/SplashActivity.kt
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317443378>:
>
> > @@ -0,0 +1,27 @@
> +package org.oppia.app.splash
> +
> +import android.os.Bundle
> +import android.view.WindowManager
> +import androidx.appcompat.app.AppCompatActivity
> +import org.oppia.app.R
> +import android.content.res.Configuration
> +
> +/*
> +* This activity acts as a container to SplashFragment
>
> Nit: please keep sentences next to each other & align the *s. Also,
> Javadocs begin with /**. Finally, we should describe what the activity
> contains, rather than making it related to the fragment (since the fragment
> is really just an implementation detail). Suggested change:
>
> /** * An activity that shows a temporary loading page until the app is fully opened. */
>
> Though, if there are no additional details to include that that can be
> simplified to:
>
> /** An activity that shows a temporary loading page until the app is fully opened. */
>
> ------------------------------
>
> In app/src/main/java/org/oppia/app/splash/SplashActivity.kt
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317443876>:
>
> > +package org.oppia.app.splash
> +
> +import android.os.Bundle
> +import android.view.WindowManager
> +import androidx.appcompat.app.AppCompatActivity
> +import org.oppia.app.R
> +import android.content.res.Configuration
> +
> +/*
> +* This activity acts as a container to SplashFragment
> +* There is a background theme set to this activity to display app logo until the app is initialized
> +*/
> +class SplashActivity : AppCompatActivity() {
> +
> +    override fun onCreate(savedInstanceState: Bundle?) {
> +        super.onCreate(savedInstanceState)
>
> Although we don't yet have a style guide outlined in the overall
> architecture document, let's follow a hybrid of JetBrain's Kotlin style
> guide and Google's Java style guide. In this case, per
> https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation
> let's use 2 spaces for indentation and 4 spaces for continuation (which I
> think I configured at the project level for Kotlin--maybe make sure that
> you're using the project configuration for Kotlin in your IDE and try
> reformatting the code via the IDE).
>
> We can evaluate later whether this is the spacing that we want, though I'm
> inclined to try and fit more code on a line rather than increasing
> whitespace since it makes it harder to read on smaller screens.
> ------------------------------
>
> In app/src/main/java/org/oppia/app/splash/SplashActivity.kt
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317443969>:
>
> > +
> +/*
> +* This activity acts as a container to SplashFragment
> +* There is a background theme set to this activity to display app logo until the app is initialized
> +*/
> +class SplashActivity : AppCompatActivity() {
> +
> +    override fun onCreate(savedInstanceState: Bundle?) {
> +        super.onCreate(savedInstanceState)
> +        setContentView(R.layout.activity_splash)
> +
> +        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
> +
> +        val fragment = SplashFragment()
> +        val transaction = supportFragmentManager.beginTransaction()
> +        transaction.replace(R.id.fragment_container, fragment)
>
> Small nit: these four lines can be done in one statement. Let's do that
> instead of splitting it up.
> ------------------------------
>
> In app/src/main/java/org/oppia/app/splash/SplashFragment.kt
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317444094>:
>
> > @@ -0,0 +1,45 @@
> +package org.oppia.app.splash
> +
> +import android.content.Intent
> +import android.os.Bundle
> +import android.view.LayoutInflater
> +import android.view.View
> +import android.view.ViewGroup
> +import androidx.fragment.app.Fragment
> +import org.oppia.app.HomeActivity
> +import org.oppia.app.R
> +
> +/*
> +* The SplashFragments navigates to Home page once
>
> This should be phrased in terms of its type, e.g.:
>
> A fragment that navigates to HomeActivity once...
>
> Please update, and format the Javadoc bits like mentioned above in
> HomeActivity.
> ------------------------------
>
> In app/src/main/java/org/oppia/app/splash/SplashFragment.kt
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317444187>:
>
> > +package org.oppia.app.splash
> +
> +import android.content.Intent
> +import android.os.Bundle
> +import android.view.LayoutInflater
> +import android.view.View
> +import android.view.ViewGroup
> +import androidx.fragment.app.Fragment
> +import org.oppia.app.HomeActivity
> +import org.oppia.app.R
> +
> +/*
> +* The SplashFragments navigates to Home page once
> +* the app is finished loading completely
> +* */
> +
>
> Nit: Javadocs should not have a blank space between them and the
> class/method/field that they're documenting.
> ------------------------------
>
> In app/src/main/java/org/oppia/app/splash/SplashFragment.kt
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317444225>:
>
> > +import androidx.fragment.app.Fragment
> +import org.oppia.app.HomeActivity
> +import org.oppia.app.R
> +
> +/*
> +* The SplashFragments navigates to Home page once
> +* the app is finished loading completely
> +* */
> +
> +class SplashFragment : Fragment() {
> +
> +    override fun onCreate(savedInstanceState: Bundle?) {
> +        super.onCreate(savedInstanceState)
> +
> +    }
> +    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
>
> Nit: please add spacing between methods.
> ------------------------------
>
> In app/src/main/java/org/oppia/app/splash/SplashFragment.kt
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317444330>:
>
> > +import org.oppia.app.R
> +
> +/*
> +* The SplashFragments navigates to Home page once
> +* the app is finished loading completely
> +* */
> +
> +class SplashFragment : Fragment() {
> +
> +    override fun onCreate(savedInstanceState: Bundle?) {
> +        super.onCreate(savedInstanceState)
> +
> +    }
> +    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
> +                              savedInstanceState: Bundle?): View? {
> +        // Inflate the layout for this fragment
>
> For this & the next comment (close this activity). The code seems pretty
> self-documenting in both places, so these comments seem unnecessary.
> ------------------------------
>
> In app/src/main/java/org/oppia/app/splash/SplashFragment.kt
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317444529>:
>
> > +import androidx.fragment.app.Fragment
> +import org.oppia.app.HomeActivity
> +import org.oppia.app.R
> +
> +/*
> +* The SplashFragments navigates to Home page once
> +* the app is finished loading completely
> +* */
> +
> +class SplashFragment : Fragment() {
> +
> +    override fun onCreate(savedInstanceState: Bundle?) {
> +        super.onCreate(savedInstanceState)
> +
> +    }
> +    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
>
> Nit on continuation: prefer continuing at the higher syntactical element
> (e.g. '(' rather than ',') per
> https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break
> .
> ------------------------------
>
> In app/src/main/java/org/oppia/app/splash/SplashFragment.kt
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317444563>:
>
> > +        routeToHomePage();
> +
> +        return view
> +    }
> +    private fun routeToHomePage() {
> +
> +        val intent = Intent(requireContext(), HomeActivity::class.java)
> +        startActivity(intent)
> +
> +        // close this activity
> +        activity!!.finish()
> +
> +    }
> +
> +}
> +
>
> Nit: let's have just one EOF line terminator. Ditto elsewhere.
> ------------------------------
>
> In app/src/main/res/layout/activity_splash.xml
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317446211>:
>
> > @@ -0,0 +1,9 @@
> +<?xml version="1.0" encoding="utf-8"?>
> +<androidx.constraintlayout.widget.ConstraintLayout
>
> I think you mentioned earlier that this was changed away from a
> ConstraintLayout, but it seems like it's still one--can we make this a
> FrameLayout, instead?
> ------------------------------
>
> In app/src/main/res/values/dimens.xml
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317446273>:
>
> > @@ -0,0 +1,5 @@
> +<?xml version="1.0" encoding="utf-8"?>
> +<resources>
> +  <integer name="fade_in_duration">1000</integer>
>
> Please include the time unit in the name, e.g. fade_in_duration_ms since
> this appears to be in milliseconds.
> ------------------------------
>
> In app/src/main/res/values/styles.xml
> <https://github.com/oppia/oppia-android/pull/60#discussion_r317446545>:
>
> > @@ -5,4 +5,25 @@
>      <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
>      <item name="colorAccent">@color/colorAccent</item>
>    </style>
> +
> +  <style name="OppiaThemeWithoutActionBar" parent="Theme.AppCompat.Light.NoActionBar">
> +    <item name="colorPrimary">@color/colorPrimary</item>
> +    <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
> +    <item name="colorAccent">@color/colorAccent</item>
> +  </style>
> +
> +  <style name="SplashScreenTheme" parent="OppiaThemeWithoutActionBar">
> +    <item name="android:windowBackground">@drawable/bg_splash</item>
> +    <item name="android:windowAnimationStyle">@style/Animation.Activity</item>
> +    <item name="android:backgroundDimEnabled">true</item>
>
> Out of curiosity, why do we want to enable this?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia-android/pull/60?email_source=notifications&email_token=AD4LXZG3VBTDX66AWCHU6ATQGNLBBA5CNFSM4IOCA73KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCCTR7PA#pullrequestreview-279388092>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD4LXZE4MO3MDOD2HHRNDGTQGNLBBANCNFSM4IOCA73A>
> .
>

veena14cs committed:
Update styles.xml

veena14cs committed:
fixed java doc comments

veena14cs committed:
Update activity_splash.xml

Changed to framelayout

veena14cs committed:
Fixed issues

veena14cs committed:
uncommented some code

veena14cs committed:
fixed .idea file

BenHenning commented:
Please reassign once this is ready to review again.

veena14cs committed:
writting unit test

veena14cs committed:
Workinh on writting testcase for fragment

veena14cs committed:
Merge branch 'splash-screen' of https://github.com/oppia/oppia-android into splash-screen

veena14cs committed:
Fixing issue with test case

veena14cs committed:
uncommented code

veena14cs committed:
Merge branch 'develop' into splash-screen

veena14cs committed:
Merge branch 'develop' into splash-screen

BenHenning commented on .idea/kotlinc.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Why is this file needed? If we don't need it, let's exclude it.

BenHenning commented on app/build.gradle:
```diff
@@ -65,6 +65,7 @@ dependencies {
       'androidx.test:runner:1.2.0',
       'com.google.truth:truth:0.43',
   )
+  debugImplementation 'androidx.fragment:fragment-testing:1.2.0-alpha02'
```
Why do we need this as a debug dependency--isn't this a test-only dependency?

veena14cs replied:
it is removed

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashActivity.kt:
```diff
@@ -0,0 +1,23 @@
+import androidx.appcompat.app.AppCompatActivity
+import org.oppia.app.R
+import android.content.res.Configuration
+/** An activity that shows a temporary loading page until the app is fully opened. */
```
Please put spacing between the imports and the Javadoc.

veena14cs replied:
fixed spacing

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashActivity.kt:
```diff
@@ -0,0 +1,23 @@
+import android.view.WindowManager
+import androidx.appcompat.app.AppCompatActivity
+import org.oppia.app.R
+import android.content.res.Configuration
```
Please order imports alphabetically. This should go between Bundle and WindowManager I believe.

BenHenning replied:
That being said, is this import actually used? If not, please remove it.

veena14cs replied:
Configuration import is not being used. Will clear it

BenHenning commented on app/src/main/res/layout/fragment_splash.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
```
For this & the corresponding activity file, let's name them similarly to their Kotlin files. E.g. ``splash_activity`` and ``splash_fragment``. This should help with naming consistency.

veena14cs replied:
Okay will correct them.

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,28 @@
+
+  @Test
+  fun splashContainerIsDisplayed() {
+    launchFragment<SplashFragment>()
```
Given that we're using activity scenario rule with SplashActivity, I don't think this is necessary. We can test splash fragment directly through the activity. E.g.:

```kotlin
@Test
fun testSplashFragment_isDisplayed() {
  onView(withId(R.id.fragment_container)).check(matches(isDisplayed()));
}
```

veena14cs replied:
Corrected.  I have changed the code

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,28 @@
+
+/** Tests for [SplashActivity]. */
+@RunWith(AndroidJUnit4::class)
+class SplashActivityTest {
```
Another worthwhile test: making sure the HomeActivity is intented to. That apparently can be done with an additional Espresso library for intenting: https://stackoverflow.com/a/47517342/3689782.

Between that and the test below, those seem sufficient to me for the splash activity.

veena14cs replied:
but its not passing testcase. It is giving error saying `androidx.test.espresso.NoMatchingViewException: No views in hierarchy found matching: with id: org.oppia.app:id/fragment_container`

BenHenning reviewed (changes requested):
@veena14cs did another pass on this, including some comments on tests. I also noticed there were a bunch of build artifacts included in this PR--please make sure those aren't included & remove the current ones. We shouldn't be checking in any build artifacts.

veena14cs committed:
Fixed spacing, rename layout files

veena14cs committed:
working on test case

veena14cs committed:
issue in passing test case

droidizer commented on app/src/main/java/org/oppia/app/splash/SplashFragment.kt:
```diff
@@ -0,0 +1,34 @@
+    savedInstanceState: Bundle?
+  ): View? {
+    val view = inflater.inflate(R.layout.splash_fragment, container, false)
+    routeToHomePage();
```
Call this function onCreate() or remove onCreate(). Also, there's a semicolon in the function call :P

veena14cs replied:
resolved

veena14cs committed:
Update SplashFragment.kt

veena14cs committed:
Update SplashFragment.kt

seanlip commented:
@veena14cs -- are you still working on this, or is it ready for review? If the latter, please make sure to update the Assignees field accordingly. Thanks!

veena14cs commented:
> @veena14cs -- are you still working on this, or is it ready for review? If the latter, please make sure to update the Assignees field accordingly. Thanks!

@seanlip I was facing issues in testcases. Its not passing testcase. Once it is done I will assign.

veena14cs committed:
working on testcase

BenHenning commented:
@veena14cs although this is lower priority than content card work, can we try to get this in a state where it's ready for review again? We should just try to get this submitted.

Also, can you clarify what you're blocked on regarding testing? I may be able to help.

veena14cs committed:
integrated testcase

veena14cs commented:
> @veena14cs although this is lower priority than content card work, can we try to get this in a state where it's ready for review again? We should just try to get this submitted.
> 
> Also, can you clarify what you're blocked on regarding testing? I may be able to help.

I have written code for testing and it is passing test case successfully. But after passing it closes `HomeActivity` without throwing any error. PTAL

veena14cs committed:
Delete kotlinc.xml

veena14cs committed:
Update build.gradle

veena14cs committed:
Update AndroidManifest.xml

veena14cs commented:
> > @veena14cs although this is lower priority than content card work, can we try to get this in a state where it's ready for review again? We should just try to get this submitted.
> > Also, can you clarify what you're blocked on regarding testing? I may be able to help.
> 
> I have written code for testing and it is passing test case successfully. But after passing it closes `HomeActivity` without throwing any error. PTAL

This PR is ready to merge PTAL

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashActivity.kt:
```diff
@@ -0,0 +1,18 @@
+    setContentView(R.layout.splash_activity)
+
+    getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
+    supportFragmentManager.beginTransaction().add(R.id.fragment_container, SplashFragment()).commitNow()
```
I suggest first checking that SplashFragment is not already present, otherwise it will be readded on each rotation.

This could be done via like so:

```kotlin
val splashFragment = supportFragmentManager.findFragmentById(R.id.fragment_container) as SplashFragment
if (splashFragment == null) {
  supportFragmentManager.beginTransaction()...
}
```

veena14cs replied:
Ok done

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashFragment.kt:
```diff
@@ -0,0 +1,30 @@
+import org.oppia.app.HomeActivity
+import org.oppia.app.R
+
+/** This SplashFragments navigates to Home page once the app is finished loading completely*/
```
Nit: There should be no empty whitespace between Javadocs and classes, Javadocs should end with punctuation, and there should be a space between the last symbol and '*/'. Also, class Javadocs should read like "An object that does X" or "A type for Y". See: https://google.github.io/styleguide/javaguide.html#s7.1-javadoc-formatting.

One way to change this:

```kotlin
/** A fragment that navigates to [HomeActivity] once the app is fully loaded. */
class SplashFragment...
```

veena14cs replied:
Ok will follow it.

BenHenning commented on domain/build/generated/source/buildConfig/debug/org/oppia/domain/BuildConfig.java:
```diff
@@ -0,0 +1,13 @@
+/**
+ * Automatically generated file. DO NOT MODIFY
```
Why do we need this file? I haven't seen it generated yet. If we don't need it, can we remove it?

veena14cs replied:
It was auto generated. Its been removed now

BenHenning commented on domain/build/intermediates/aapt_friendly_merged_manifests/debug/aapt/AndroidManifest.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
```
it is auto generating even after deleting

BenHenning commented on domain/build/intermediates/aapt_friendly_merged_manifests/debug/aapt/output.json:
```diff
@@ -0,0 +1 @@
+[{"outputType":{"type":"AAPT_FRIENDLY_MERGED_MANIFESTS"},"apkData":{"type":"MAIN","splits":[],"versionCode":1,"versionName":"1.0","enabled":true,"outputFile":"domain-debug.aar","fullName":"debug","baseName":"debug"},"path":"AndroidManifest.xml","properties":{"packageId":"org.oppia.domain","split":""}}]
```
Ditto for this file--let's remove it if we don't need it.

veena14cs replied:
it is auto generating even after deleting

veena14cs replied:
> Also per your earlier comment, we discussed via chat yesterday that the test should close right away after completing if it passes or fails, and if Android Studio is indicating that it's passing then everything should be working per the test code.

Yes, it passes testcase sucessfully and closes it after completion.

BenHenning commented on domain/build/intermediates/library_manifest/debug/AndroidManifest.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Ditto for this file--let's remove it if we don't need it.

BenHenning replied:
Following up on this--please remove all build artifacts from the domain module.

BenHenning commented on domain/build/intermediates/merged_manifests/debug/output.json:
```diff
@@ -0,0 +1 @@
+[{"outputType":{"type":"MERGED_MANIFESTS"},"apkData":{"type":"MAIN","splits":[],"versionCode":1,"versionName":"1.0","enabled":true,"outputFile":"domain-debug.aar","fullName":"debug","baseName":"debug"},"path":"../../library_manifest/debug/AndroidManifest.xml","properties":{"packageId":"org.oppia.domain","split":""}}]
```
Ditto for this file--let's remove it if we don't need it.

BenHenning commented on domain/build/outputs/logs/manifest-merger-debug-report.txt:
```diff
@@ -0,0 +1,37 @@
+-- Merging decision tree log ---
```
Ditto for this file--let's remove it if we don't need it.

veena14cs replied:
deleted

BenHenning commented on model/build.gradle:
```diff
@@ -23,7 +23,9 @@ protobuf {
-  implementation 'com.google.protobuf:protobuf-lite:3.0.0'
+  // This must use compile in order for lite protobufs to generate correctly.
+  // TODO(#59): Move to Bazel to avoid relying on a deprecated Gradle directive.
+  compile 'com.google.protobuf:protobuf-lite:3.0.0'
```
Is this change still needed after you merge in the latest develop branch? I'm surprised that this would be needed since the splash page doesn't use any protos.

BenHenning commented on app/src/main/java/org/oppia/app/splash/SplashFragment.kt:
```diff
@@ -0,0 +1,30 @@
+    inflater: LayoutInflater, container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    val view = inflater.inflate(R.layout.splash_fragment, container, false)
```
Do we actually need a view for this fragment? If we're just using the content view of the activity for the transition, let's remove the extra splash fragment layout since it's not being used for anything.

BenHenning replied:
Also, are we still setting the splash page drawable anywhere? I'm not noticing it.

BenHenning replied:
@veena14cs can you address this comment?

veena14cs replied:
Since it was design from Technical Document's perspective I had added fragment but as its no where used I am removing it

veena14cs replied:
> Also, are we still setting the splash page drawable anywhere? I'm not noticing it.

splash page was earlier used through custom drawable file `bg_splash.xml` now I am setting it directly to style.

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,47 @@
+class SplashActivityTest {
+
+  @get:Rule
+  var mActivityRule: IntentsTestRule<SplashActivity> = IntentsTestRule(
```
Prefer not prefixing fields with 'm', 's', etc. that AOSP code typically does. This isn't conventional for regular Java/Kotlin code, and is a bit verbose given that it's clear from context which variables are members/statics.

veena14cs replied:
corrected it.

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,47 @@
+    Assert.assertNotNull(mainActivity)
+  }
+
+  private fun startFragment(fragment: Fragment) {
```
Why is this needed? If we're creating SplashActivity directly, shouldn't that already have splash fragment created & added?

veena14cs replied:
It throws an error 
`androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: Wanted to match 1 intents. Actually matched 0 intents.`

`IntentMatcher: has component: has component with: class name: is` `"org.oppia.app.HomeActivity" package name: an instance of java.lang.String short` `class name: an instance of java.lang.String`
if I don't start a fragment over here

veena14cs replied:
I have made use of ActivityTestRule<SplashActivity>. I was assuming that I need to launch fragment and then it will launch HomeActivity. 

It is passing testcase now. Code is updated

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,47 @@
+  }
+
+  @Test
+  fun testMainActivity() {
```
Test names should include both the condition being tested and the expected outcome. Here, I suppose we want to verify that the routing is working correctly. Consider:

``testSplashActivity_initialOpen_routesToHomeActivity``

In this case, the action/unit being tested is SplashActivity, the condition is 'initialOpen', and the outcome should be 'routesToHomeActivity'.

veena14cs replied:
Thanks for this note. Corrected it.

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,47 @@
+
+  @Test
+  fun testMainActivity() {
+    Assert.assertNotNull(mainActivity)
```
Shouldn't we verify via ``IntentsTestRule`` that the correct activity was intented to?

veena14cs replied:
made changes in the code

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,47 @@
+class SplashActivityTest {
+
+  @get:Rule
+  var mActivityRule: IntentsTestRule<SplashActivity> = IntentsTestRule(
```
Does ``IntentsTestRule`` work correctly when running this as a Robolectric test instead of Espresso?

veena14cs replied:
Made use of ActivityTestRule<SplashActivity> to launch activity. Code is changed

BenHenning reviewed (changes requested):
Thanks @veena14cs! Left a few comments.

BenHenning commented:
Also per your earlier comment, we discussed via chat yesterday that the test should close right away after completing if it passes or fails, and if Android Studio is indicating that it's passing then everything should be working per the test code.

veena14cs committed:
fixed issues

veena14cs committed:
Fixed testcases

veena14cs committed:
removed unused imports

BenHenning commented on .idea/misc.xml:
```diff
@@ -6,4 +6,4 @@
     <option name="id" value="Android" />
   </component>
-</project>
+</project>
```
Please revert this file--let's keep line endings at the end of XML files for consistency.

veena14cs replied:
reverted

BenHenning commented on app/build.gradle:
```diff
@@ -56,14 +56,15 @@ dependencies {
       'androidx.test.ext:junit:1.1.1',
       'com.google.truth:truth:0.43',
-      'org.robolectric:robolectric:4.3',
+      'org.robolectric:robolectric:4.3'
```
Super minor nit here & below: prefer keeping the trailing comma just to make it easier to add other dependencies later (or move them around).

veena14cs replied:
added comma

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,43 @@
+  }
+
+  @Test
+  @Throws(InterruptedException::class)
```
This isn't needed since this is a Kotlin test (Kotlin doesn't have explicit ``throws`` declarations on methods).

veena14cs replied:
removed exception

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,43 @@
+  @Test
+  @Throws(InterruptedException::class)
+  fun testSplashActivity_initialOpen_routesToHomeActivity() {
+
```
Nit: please remove blank newline at the beginning of the method for consistency with other checked in code.

veena14cs replied:
removed blank lines

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,43 @@
+
+  @get:Rule
+  var activityTestRule: ActivityTestRule<SplashActivity> = ActivityTestRule(
+    SplashActivity::class.java, true,
```
Out of curiosity, why are you changing the default launch/touch mode values? Can you provide a brief line comment here explaining that?

Also, prefer documenting the names of these parameters to help folks reading the code outside of Android Studio, e.g.:

```kotlin
@get:Rule
var activityTestRule: ActivityTestRule<SplashActivity> =
    ActivityTestRule(SplashActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false)
```

veena14cs replied:
Given comments and provided documentation links for each parameter

BenHenning replied:
Please update the code to include in-line comments naming the parameters. The documentation above doesn't replace the inline-comments. These are still beneficial since they provide context for the literal values.

BenHenning replied:
@veena14cs this is still unresolved. Note that when I provide code suggestions, I suggest actually directly applying them since that's specifically what I'm looking for.

BenHenning reviewed (changes requested):
Thanks @veena14cs! This LGTM except for the one comment I have open about the potentially unused splash page drawable.

BenHenning commented:
@veena14cs can you please update this branch to base off of ``develop`` and update it with the latest changes?

veena14cs committed:
added comma for dependency

veena14cs committed:
Merge branch 'develop' into splash-screen

veena14cs commented:
> @veena14cs can you please update this branch to base off of `develop` and update it with the latest changes?

updated to latest changes and changed base branch to develop.

veena14cs committed:
fixed minor issues

veena14cs committed:
added javadoc comments for activitytestrule

veena14cs committed:
removed blank spaces

veena14cs committed:
removed splashFragment as its not used anywhere

veena14cs committed:
Delete splash_fragment.xml

veena14cs committed:
deleted bg_splash

veena14cs committed:
Update build.gradle

veena14cs committed:
removed blank spaces

veena14cs committed:
Update SplashActivityTest.kt

BenHenning commented on app/src/main/res/values/styles.xml:
```diff
@@ -6,4 +6,24 @@
+  <style name="Animation" />
+
+  <!-- Standard animations for a full-screen window or activity. -->
+  <style name="Animation.Activity">
```
Maybe instead call this FadeTransitionedActivity or something similar. Not sure if we want this to be the default for all activities, and it will be clearer for each activity that uses this what the style does if we include the fade bit in the name.

veena14cs replied:
Changed to FadeTransitionAnimation

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,59 @@
+  /**
+   * Creates an {@link ActivityTestRule} for the Activity under test.
+   *
+   * @param SplashActivity    The activity under test. This must be a class in the instrumentation
```
This actually wasn't what I was looking for--in fact, we don't want to just copy the documentation explanations (please remove).

What I meant below by adding a brief comment explaining why we do this is along the lines of:

```kotlin
// This enables initialTouchMode because of X. This disables the default launchActivity parameter because of Y.
@get:Rule
...
```

That helps provide context to other developers on why these values were chosen, and why they're important.

veena14cs replied:
@get:Rule
  var activityTestRule: ActivityTestRule<SplashActivity> = ActivityTestRule(
    SplashActivity::class.java, true,// initialTouchMode. True to launch activity in touch mode
    false // launchActivity. False to set intent per test
  )

Can we write comment in this way in kotlin

BenHenning replied:
Unfortunately, this probably won't wrap correctly when using a code reformatter, which is why I suggest the ``/* name= */`` style. Also, it's more concise. We can explain why specific literal values were chosen in a comment above the variable, and only describe the parameter name in an in-line comment.

To be clear, there are two different things we're documenting here: what the parameter is the literal value is corresponding to, and why the specific value chosen was chosen. The former should be an in-line comment, and the latter should be documentation near the variable.

veena14cs replied:
resolved comment

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,59 @@
+
+  @Before
+  fun setUp() {
+    //initial setup code
```
Nit here & below for the tear-down method: these comments aren't really adding any value since the context implies that the methods are initial setup/clean up code blocks, respectively. I suggest removing these comments.

veena14cs replied:
ok removed

BenHenning reviewed (changes requested):
Thanks @veena14cs--I have some concerns about the documentation you added to the test. Please take a look.

Also, can we remove the build artifact files before submitting the PR? We shouldn't check in these files. I saw your earlier comments mention these files keep coming back as well. Can you make sure your .gitignore file is ignoring changes under build/ folders? My .gitignore file includes the following filter to prevent these files from showing up in my staged changes:

``domain/build``

veena14cs committed:
Update SplashActivityTest.kt

veena14cs committed:
Delete AndroidManifest.xml

veena14cs committed:
Merge branch 'develop' into splash-screen

veena14cs committed:
Update styles.xml

veena14cs committed:
Update SplashActivityTest.kt

veena14cs committed:
Delete output.json

veena14cs committed:
Delete manifest-merger-debug-report.txt

veena14cs committed:
Update SplashActivityTest.kt

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,42 @@
+@RunWith(AndroidJUnit4::class)
+class SplashActivityTest {
+
+  // The initialTouchMode enables the activity to be launched in touch mode.The launchActivity is disabled to launch Activity explicitly within each test case
```
Please wrap this according to the line wrapping:

```kotlin
  // The initialTouchMode enables the activity to be launched in touch mode. The launchActivity is
  // disabled to launch Activity explicitly within each test case.
```

veena14cs replied:
resolved comment issue

BenHenning reviewed (approved):
Thanks @veena14cs. Everything looks fine here, but I'd like you to resolve the open comment thread before merging. Please follow the format I suggested where you use an inline comment like ``/* paramName= */`` next to literal values, and document why we chose those values in single line, line-wrapped, comments above the field.

veena14cs committed:
Update SplashActivityTest.kt

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,42 @@
+import org.junit.runner.RunWith
+import org.oppia.app.home.HomeActivity
+
+/** Tests for [SplashActivity].
```
Nit: we should wrap this documentation as described in https://google.github.io/styleguide/javaguide.html#s7.1.1-javadoc-multi-line if it spans multiple lines:

```kotlin
/**
 * Tests for [SplashActivity]. For context on the activity test rule setup see:
 * https://jabknowsnothing.wordpress.com/2015/11/05/activitytestrule-espressos-test-lifecycle/.
 */
```

veena14cs replied:
ok corrected. Will follow this.

BenHenning commented on app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt:
```diff
@@ -0,0 +1,42 @@
+class SplashActivityTest {
+
+  @get:Rule
+  var activityTestRule: ActivityTestRule<SplashActivity> = ActivityTestRule(
```
Given that we've gone back and forth on this several times, I feel like I wasn't clear enough. Please use the following formatting for this rule per previous comment threads:

```kotlin
  // The initialTouchMode enables the activity to be launched in touch mode. The launchActivity is
  // disabled to launch Activity explicitly within each test case.
  @get:Rule
  var activityTestRule: ActivityTestRule<SplashActivity> = ActivtyTestRule(
    SplashActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false)
```

veena14cs replied:
Thanks! I assumed it other way around.

BenHenning reviewed (approved):
Thanks @veena14cs. Please apply the two code snippets and resolve the conflict, and then this is good to submit.

In the future, I suggest copying suggested code snippets from comments directly into your code, and if they seem like they might not work or don't make sense follow up with a question to help narrow down a solution between you and the reviewer.

veena14cs committed:
Update SplashActivityTest.kt

veena14cs commented:
> Thanks @veena14cs. Please apply the two code snippets and resolve the conflict, and then this is good to submit.
> 
> In the future, I suggest copying suggested code snippets from comments directly into your code, and if they seem like they might not work or don't make sense follow up with a question to help narrow down a solution between you and the reviewer.



> Thanks @veena14cs. Please apply the two code snippets and resolve the conflict, and then this is good to submit.
> 
> In the future, I suggest copying suggested code snippets from comments directly into your code, and if they seem like they might not work or don't make sense follow up with a question to help narrow down a solution between you and the reviewer.

Thanks @BenHenning will keep this in mind. I have copied these 2 comments and I have I updated code. Its ready to merge.

veena14cs committed:
updated manifest

veena14cs committed:
Update AndroidManifest.xml

veena14cs commented:
> Thanks @veena14cs. Please apply the two code snippets and resolve the conflict, and then this is good to submit.
> 
> In the future, I suggest copying suggested code snippets from comments directly into your code, and if they seem like they might not work or don't make sense follow up with a question to help narrow down a solution between you and the reviewer.

Thanks @BenHenning will keep this in mind. I have copied these 2 comments and I have I updated code. Its ready to merge.
